lib/routes.js

```diff
@@ -36,10 +36,21 @@ const create = async function (server, { api, basedir, cors, vhost, handlers, ex
 
     for (const [path, operations] of Object.entries(api.paths)) {
         const pathnames = Utils.unsuffix(path, '/').split('/').slice(1).join(SEPARATOR);
+        const hapiHandler = operations['x-hapi-handler'];
```

Comment:
the handler in case of x-hapi-handler is already set few lines before, this is probably a bug as we probably wanted precedence of x-hapi-handler over a path but now we do it twice.